### PR TITLE
 Use ProperSubtypeData instead of SubtypeData

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by ushirask** [comment]
#33756 & #34115 will make use of changes introduced here

File: semtypes/src/main/java/io/ballerina/types/Core.java [hunk 1]
```diff
@@ -120,7 +120,7 @@
                 int c = code.code;
                 all = UniformTypeBitSet.from(all.bitset | 1 << c);
             } else {
-                subtypes.add(UniformSubtype.from(code, data));
+                subtypes.add(UniformSubtype.from(code, (ProperSubtypeData) data));
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/types/Core.java:123]: Shall we retain these comments https://github.com/ballerina-platform/nballerina/commit/86ecd997d3031cc04e4e3d453cad6a5d2cabad78#diff-5008fa7817f6124661e3e735e6673f37538af2ab4de42527dc7c030ceea9afc2R519

File: semtypes/src/main/java/io/ballerina/types/Core.java [hunk 2]
```diff
@@ -286,11 +286,11 @@
                     }
                 }
             }
-            if (data instanceof AllOrNothingSubtype && ((AllOrNothingSubtype) data).isAllSubtype()) {
+            if (!(data instanceof AllOrNothingSubtype)) {
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/types/Core.java:289]: Let's add a comment to indicate that this is change to reflect the changes done for JBUG

**Review by rdhananjaya — APPROVED** [review]

**Comment by rdhananjaya** [comment]
@ushirask please update the PRs that depend on this changes

**Comment by ushirask** [comment]
#33756 & #34115 should reflect the changes depending on this now.

**Comment by ushirask** [comment]
> @ushirask please update the PRs that depend on this changes

#33756 & #34115 should reflect the changes depending on this now.

**Comment by rdhananjaya** [comment]
> > @ushirask please update the PRs that depend on this changes
> 
> #33756 & #34115 should reflect the changes depending on this now.

I don't think that's how PRs diffs work in github, you need to update your PRs with target branch.

Screenshot 2021-12-16 at 14 09 13

**Comment by ushirask** [comment]
> > > @ushirask please update the PRs that depend on this changes
> > 
> > 
> > #33756 & #34115 should reflect the changes depending on this now.
> 
> I don't think that's how PRs diffs work in github, you need to update your PRs with target branch.
> 

my bad, should be fixed now.

**Pull request merged**